app/src/main/java/nerd/tuxmobil/fahrplan/congress/utils/AlarmToneConversion.kt

```diff
@@ -0,0 +1,37 @@
+package nerd.tuxmobil.fahrplan.congress.utils
+
+import android.net.Uri
+import androidx.core.net.toUri
+
+/**
+ * Conversions (`Uri` <-> `String`) related to alarm tone data.
+ */
+internal object AlarmToneConversion {
+
+    private val SILENT_URI: Uri? = null
+    private const val SILENT_STRING: String = ""
+
+    /**
+     * Returns a nullable `Uri` to configure a notification with.
+     */
+    fun getNotificationIntentUri(alarmTone: String?, defaultValue: Uri): Uri? = when (alarmTone) {
+        null -> defaultValue
+        SILENT_STRING -> SILENT_URI
+        else -> alarmTone.toUri()
+    }
+
+    /**
+     * Returns a nullable `String` to be persisted.
+     */
+    fun getPersistableString(alarmToneUri: Uri?): String? {
+        return alarmToneUri?.toString() ?: SILENT_STRING
+    }
+
+    /**
+     * Returns a nullable `Uri` to initialize an alarm tone picker with.
+     */
+    fun getPickerIntentUri(persisted: String): Uri? = persisted.let { persistedValue ->
+        if (persistedValue == SILENT_STRING) SILENT_URI else persistedValue.toUri()
+    }
```

Comment:
The `let` is redundant now.

```suggestion
    fun getPickerIntentUri(persistedValue: String): Uri? {
        return if (persistedValue == SILENT_STRING) SILENT_URI else persistedValue.toUri()
    }
```

Reply:
:heavy_check_mark: Changed in d350abd490dc6c769b9b0cd9b4c512ad209fd314.